patroni/config.py

```diff
@@ -111,6 +111,9 @@ def __init__(self, configfile, validator=default_validator):
         self._load_cache()
         self._cache_needs_saving = False
 
+    def __get_permission_for_data_dir(self):
+        return int(oct(os.stat(self._data_dir).st_mode), base=8)
```

Comment:
I afraid it will not work on Windows

Reply:
yep, forgot about Windows case, will fix it

Reply:
actually, it works for me on Windows machine, or may be I am missing something ?

```
import os
import sys

if __name__ == '__main__':
    print(sys.platform)
    f = open("./test", "a")
    f.close()
    os.chmod("./test", 0o777)
    print(oct(os.stat("./test").st_mode))
    rights = int(oct(os.stat("./test").st_mode), base=8)
    print(rights)
    os.chmod("./test", int(oct(os.stat("./test").st_mode), base=8))
```

And the output:
```
win32
0o100666
33206
```

Reply:
Oh, true, I mixed it up with some other issue.
Anyway, we should not copy directory permissions as it is, but replicate https://github.com/postgres/postgres/blob/REL_13_STABLE/src/common/file_perm.c#L33-L50
That is, set group access only when it is necessary and by default only user should be allowed.

Reply:
@anikin-aa please take a look at my last comment https://github.com/zalando/patroni/pull/1901#discussion_r609657320

Reply:
@CyberDem0n , yep, i see. Sorry for long response, pretty busy these days. I believe, that i will comeback on the next week and continue to work on this.

Reply:
@anikin-aa ping

Reply:
@anikin-aa ping

Reply:
@CyberDem0n , sorry for long response, i will to fix it during next week

Reply:
@CyberDem0n , let me verify the requested changes (just to double check, that i understood everything correctly)
i've to check if mode on pg data dir equals to PG_DIR_MODE_GROUP, then return group file mode otherwise return file mode owner ? because right now i am changing permissions only for files

Reply:
It is a bit more complex than that.
Patroni not only creates files, but sometimes also directories (for example when copying logical slots to replicas).
In all cases umask and permissions should be set correctly and according to expectations of Postgres.
I.e., default values (owner only): https://github.com/postgres/postgres/blob/20c22333630111718bc97ce9a19b392f23e3c525/src/common/file_perm.c#L17-L25
and https://github.com/postgres/postgres/blob/20c22333630111718bc97ce9a19b392f23e3c525/src/common/file_perm.c#L37-L42 if group access is allowed.

Correct permissions should be "calculated" when:
1. We start Patroni with existing $PGDATA
2. After [bootstrap](https://github.com/zalando/patroni/blob/1b6e23ab6a5a92892c7948afde415f10cb95acdb/patroni/postgresql/bootstrap.py#L290-L294) (initdb or custom bootstrap) and after [create_replica](https://github.com/zalando/patroni/blob/1b6e23ab6a5a92892c7948afde415f10cb95acdb/patroni/postgresql/bootstrap.py#L278)

Reply:
ahh, ok, thank you for detailed explanation 

> but sometimes also directories (for example when copying logical slots to replicas).

didn't know about that, will check it too

Reply:
Hi @anikin-aa any progress on it?